api/v1alpha1/accesslogging_types.go

```diff
@@ -30,8 +30,34 @@ type ProxyAccessLogSetting struct {
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=50
 	Sinks []ProxyAccessLogSink `json:"sinks"`
+	// Scope defines the scope of component emitting the accesslog, such as envoy listener, TCP proxy, HTTP connection manager.
+	// If not defined, the setting would apply to all scopes supported by envoy gateway
+	// +kubebuilder:validation:Enum=Listener
+	// +optional
+	// +notImplementedHide
+	Scope ProxyAccessLogScopeType `json:"scope,omitempty"`
```

Comment:
* currently this only applies to route right ?
* is `Type`a more consistent term ?
* is there a way to reduce the envoyisms here by limiting the terms to `Route` and `Listener` ?

Reply:
> is there a way to reduce the envoyisms here by limiting the terms to Route and Listener ?

The best GW-API fit that I can think of is:
- `Frontend` (~ envoy listener)
- `xRoute` (HTTP/GRPCRoute ~ HCM access logs, TCP/UDPRoute ~ TCP/UDP Proxy access logs)
- `xBackend` (GRPC/HTTPBackend ~ Router access logs). Envoy currentl does
WDYT?